fsspec/spec.py

```diff
@@ -196,6 +196,17 @@ def _strip_protocol(cls, path):
         # use of root_marker to make minimum required path, e.g., "/"
         return path or cls.root_marker
 
+    def _unstrip_protocol(self, name):
+        # could be overridden for FSs that make work with URLs
+        if isinstance(self.protocol, str):
+            if name.startswith(self.protocol):
+                return name
+            return self.protocol + "://" + name
+        else:
+            if name.startswith(tuple(self.protocol)):
+                return name
+            return self.protocol[0] + "://" + name
```

Comment:
For the record: we've also implemented this before in our wrappers, e.g. https://github.com/iterative/dvc/blob/28f57933db4873b2f6fcb466e479754aa016de2f/dvc/fs/s3.py#L162